### PR TITLE
[improvement](statistics)Limit the max string length to 1024 while collecting column stats to control BE memory usage.

### PR DESCRIPTION
**Pull request by Jibing-Li**
Sometimes, one column may store a very long text value, collecting NDV for that column may use too much memory and may cause BE OOM. This pr is to limit the max string length to 1024 while collecting column stats to control BE memory usage.

## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Jibing-Li** [comment]
run buildall

**Comment by Jibing-Li** [comment]
run buildall

**Comment by Jibing-Li** [comment]
run external

File: fe/fe-core/src/main/java/org/apache/doris/statistics/BaseAnalysisTask.java [hunk 1]
```diff
@@ -98,11 +98,12 @@ public abstract class BaseAnalysisTask {
             + "${dataSizeFunction} * ${scaleFactor} AS `data_size`, "
             + "NOW() "
             + "FROM ( "
-            + "    SELECT t0.`${colName}` as `column_key`, COUNT(1) as `count` "
+            + "    SELECT t0.`colValue` as `column_key`, COUNT(1) as `count` "
             + "    FROM "
-            + "    (SELECT `${colName}` FROM `${catalogName}`.`${dbName}`.`${tblName}` ${index} "
+            + "    (SELECT SUBSTRING(CAST(`${colName}` AS STRING), 1, 1024) AS `colValue` "
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/statistics/BaseAnalysisTask.java:103]: cast all type to string?

Comment (Jibing-Li) [reply fe/fe-core/src/main/java/org/apache/doris/statistics/BaseAnalysisTask.java:103]: updated

**Comment by Jibing-Li** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Review by zddr — APPROVED** [review]

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**